src/Pages/ChannelRequests.razor

```diff
@@ -227,7 +227,13 @@
             </DataGridColumn>
             <DataGridCommandColumn TItem="ChannelOperationRequest" Caption="Actions" Displayable="true">
                 <EditCommandTemplate>
-                    <Button Color="Color.Primary" hidden=@(!_isFinanceManager) Clicked="@(() => ShowModal(context.Item))">Approve</Button>
+                    @if (_isFinanceManager && context.Item.Wallet != null && context.Item.Wallet.Keys != null 
+                         && context.Item.Wallet.Keys.Count(x => x.UserId == LoggedUser?.Id) > context.Item.ChannelOperationRequestPsbts.Count(x => x.UserSignerId == LoggedUser?.Id) 
+                         && !context.Item.AreAllRequiredHumanSignaturesCollected)
+                    {
+                        <Button Color="Color.Primary" hidden=@(!_isFinanceManager) Clicked="@(() => ShowModal(context.Item))">Approve</Button>
+                    }
+                    
```

Comment:
isn't this the job of `GetUnsignedPendingRequestsByUser`?

Reply:
I was doing it the same way as in withdrawals. You are right!

Reply:
I will do it and implement something similar for withdrawals for unifying both withdrawals and channel requests

Reply:
Done 🏁